File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoButtonDemoPage.cs
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_button_demo_page.dart
// (exact sample parity)

public sealed class CupertinoButtonDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoButtonDemoPageState();
}

internal sealed class CupertinoButtonDemoPageState : State
{
    private int _taps;
    private int _longPresses;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Cupertino buttons", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Plain, tinted and filled styles across the three size styles.",
                    fontSize: 14.0,
                    color: Color.Parse("#8A000000")),
                new Wrap(
                    spacing: 12.0,
                    runSpacing: 12.0,
                    crossAxisAlignment: WrapCrossAlignment.Center,
                    children:
                    [
                        new CupertinoButton(
                            child: new Text("Plain"),
                            onPressed: () => SetState(() => _taps++)),
                        CupertinoButton.Tinted(
                            child: new Text("Tinted"),
                            onPressed: () => SetState(() => _taps++)),
                        CupertinoButton.Filled(
                            child: new Text("Filled"),
                            onPressed: () => SetState(() => _taps++)),
                        new CupertinoButton(child: new Text("Disabled"), onPressed: null),
                        CupertinoButton.Filled(child: new Text("Disabled filled"), onPressed: null),
                    ]),
                new Text("Size styles", fontSize: 14.0, color: Colors.Black),
                new Wrap(
                    spacing: 12.0,
                    runSpacing: 12.0,
                    crossAxisAlignment: WrapCrossAlignment.Center,
                    children:
                    [
                        CupertinoButton.Filled(
                            child: new Text("Small"),
                            sizeStyle: CupertinoButtonSize.Small,
                            onPressed: () => SetState(() => _taps++)),
                        CupertinoButton.Filled(
                            child: new Text("Medium"),
                            sizeStyle: CupertinoButtonSize.Medium,
                            onPressed: () => SetState(() => _taps++)),
                        CupertinoButton.Filled(
                            child: new Text("Large"),
                            onPressed: () => SetState(() => _taps++)),
                    ]),
                new Text("Customisation", fontSize: 14.0, color: Colors.Black),
                new Wrap(
                    spacing: 12.0,
                    runSpacing: 12.0,
                    crossAxisAlignment: WrapCrossAlignment.Center,
                    children:
                    [
                        CupertinoButton.Tinted(
                            child: new Text("Grey tint"),
                            color: CupertinoColors.SystemGrey,
                            onPressed: () => SetState(() => _taps++)),
                        CupertinoButton.Filled(
                            child: new Text("Custom radius"),
                            color: CupertinoColors.SystemRed,
                            borderRadius: BorderRadius.Circular(4.0),
                            onPressed: () => SetState(() => _taps++)),
                        new CupertinoButton(
                            child: new Icon(CupertinoIcons.Heart),
                            foregroundColor: CupertinoColors.SystemPink.Value,
                            onPressed: () => SetState(() => _taps++)),
                        new CupertinoButton(
                            child: new Text("Long press me"),
                            onPressed: null,
                            onLongPress: () => SetState(() => _longPresses++)),
                    ]),
                new Text($"Taps: {_taps}   Long presses: {_longPresses}", fontSize: 14.0, color: Colors.Black),
            ]);
    }
}

File: dart_sample/lib/demos/cupertino/cupertino_button_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoButtonDemoPage extends StatefulWidget {
  const CupertinoButtonDemoPage({super.key});

  @override
  State<CupertinoButtonDemoPage> createState() => _CupertinoButtonDemoPageState();
}

class _CupertinoButtonDemoPageState extends State<CupertinoButtonDemoPage> {
  int _taps = 0;
  int _longPresses = 0;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Cupertino buttons',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Plain, tinted and filled styles across the three size styles.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Wrap(
          spacing: 12,
          runSpacing: 12,
          crossAxisAlignment: WrapCrossAlignment.center,
          children: <Widget>[
            CupertinoButton(
              onPressed: () => setState(() => _taps++),
              child: const Text('Plain'),
            ),
            CupertinoButton.tinted(
              onPressed: () => setState(() => _taps++),
              child: const Text('Tinted'),
            ),
            CupertinoButton.filled(
              onPressed: () => setState(() => _taps++),
              child: const Text('Filled'),
            ),
            const CupertinoButton(onPressed: null, child: Text('Disabled')),
            const CupertinoButton.filled(
              onPressed: null,
              child: Text('Disabled filled'),
            ),
          ],
        ),
        const Text('Size styles', style: TextStyle(fontSize: 14, color: Colors.black)),
        Wrap(
          spacing: 12,
          runSpacing: 12,
          crossAxisAlignment: WrapCrossAlignment.center,
          children: <Widget>[
            CupertinoButton.filled(
              sizeStyle: CupertinoButtonSize.small,
              onPressed: () => setState(() => _taps++),
              child: const Text('Small'),
            ),
            CupertinoButton.filled(
              sizeStyle: CupertinoButtonSize.medium,
              onPressed: () => setState(() => _taps++),
              child: const Text('Medium'),
            ),
            CupertinoButton.filled(
              onPressed: () => setState(() => _taps++),
              child: const Text('Large'),
            ),
          ],
        ),
        const Text('Customisation', style: TextStyle(fontSize: 14, color: Colors.black)),
        Wrap(
          spacing: 12,
          runSpacing: 12,
          crossAxisAlignment: WrapCrossAlignment.center,
          children: <Widget>[
            CupertinoButton.tinted(
              color: CupertinoColors.systemGrey,
              onPressed: () => setState(() => _taps++),
              child: const Text('Grey tint'),
            ),
            CupertinoButton.filled(
              color: CupertinoColors.systemRed,
              borderRadius: BorderRadius.circular(4),
              onPressed: () => setState(() => _taps++),
              child: const Text('Custom radius'),
            ),
            CupertinoButton(
              foregroundColor: CupertinoColors.systemPink,
              onPressed: () => setState(() => _taps++),
              child: const Icon(CupertinoIcons.heart),
            ),
            CupertinoButton(
              onPressed: null,
              onLongPress: () => setState(() => _longPresses++),
              child: const Text('Long press me'),
            ),
          ],
        ),
        Text(
          'Taps: $_taps   Long presses: $_longPresses',
          style: const TextStyle(fontSize: 14, color: Colors.black),
        ),
      ],
    );
  }
}
